tests/drivers/spi/spi_loopback/src/spi.c

```diff
@@ -35,15 +35,27 @@ struct spi_cs_control spi_cs = {
 #endif
 
 /* to run this test, connect MOSI pin to the MISO of the SPI */
-
 #define STACK_SIZE 512
 #define BUF_SIZE 17
+#define BUF2_SIZE 36
+
+#if CONFIG_NOCACHE_MEMORY
+uint8_t BUFFER_TX[] = "0123456789abcdef\0";
+__aligned(16) uint8_t buffer_tx[BUF_SIZE] __attribute((__section__(".nocache")));
+__aligned(16) uint8_t buffer_rx[BUF_SIZE] __attribute((__section__(".nocache.dma")));
+
+uint8_t BUFFER2_TX[] = "Thequickbrownfoxjumpsoverthelazydog\0";
+__aligned(16) uint8_t buffer2_tx[BUF2_SIZE] __attribute((__section__(".nocache")));
+__aligned(16) uint8_t buffer2_rx[BUF2_SIZE] __attribute((__section__(".nocache.dma")));
+#else
+/* Align buffers in memory to support DMA access */
 uint8_t buffer_tx[] = "0123456789abcdef\0";
-uint8_t buffer_rx[BUF_SIZE] = {};
+uint8_t __aligned(16) buffer_rx[BUF_SIZE] = {};
 
```

Comment:
Why are these changes made to the `#else` case?

Reply:
Those changes were made to ensure that the rx buffer would be aligned for DMA controllers that may require it, as DMA buffers are aligned in the DMA tests in Zephyr. After testing, the EDMA controller in the RT10xx series does not require aligned buffers, so this change has been removed